tooling/components/io.siddhi.distribution.editor.core/src/main/java/io/siddhi/distribution/editor/core/util/SourceEditorUtils.java

```diff
@@ -434,6 +439,246 @@ private static ProcessorMetaData generateProcessorMetaData(Class<?> processorCla
         }
     }
 
+    //generate the parameter data type syntax for extension syntax generation
+    private static StringBuilder parameterDataTypeGeneration(String parameterName,
+                                                             HashMap<String, DataType[]> parameterMap) {
+
+        StringBuilder parameterDataType = new StringBuilder();
+        parameterDataType.append("<");
+        DataType[] parameterType = parameterMap.get(parameterName);
+        for (int i = 0; i < parameterType.length; i++) {
+            DataType dataType = parameterType[i];
+            if (i == parameterType.length - 1) {
+                parameterDataType.append(dataType);
+            } else {
+                parameterDataType.append(dataType).append("|");
+            }
+        }
+        parameterDataType.append(">");
+        return parameterDataType;
+    }
+
+    /*generate the syntax  for functions,windows,stream processor and aggregate
+     function*/
+    private static List<SyntaxMetaData> windowFunctionSyntaxGeneration(Extension extension,
+                                                                       HashMap<String, DataType[]> parameterMap,
+                                                                       List<SyntaxMetaData> syntaxList,
+                                                                       String finalClipBoardSyntax, String finalSyntax,
+                                                                       StringBuilder syntax,
+                                                                       StringBuilder clipBoardSyntax) {
+        //Add return syntax to the extension if it is a functions and aggregate functions
+        for (ReturnAttribute returnAttribute : extension.returnAttributes()) {
+            syntax.append("<");
+            for (int j = 0; j < (returnAttribute.type()).length; j++) {
+                if (j == returnAttribute.type().length - 1) {
```

Comment:
Flip these conditions for easy readablilty
```suggestion
                if (j != returnAttribute.type().length - 1) {
```